src/Cloud/weglide/UploadFlight.cpp

```diff
@@ -58,27 +59,43 @@ MakeUploadFlightMime(CURL *easy, const WeGlideSettings &settings,
 }
 
 Co::Task<boost::json::value>
-UploadFlight(CurlGlobal &curl, const WeGlideSettings &settings,
+UploadFlight(CurlGlobal& curl, const char *url,
+             const struct Pilot &pilot,
              uint_least32_t glider_type,
              Path igc_path,
              ProgressListener &progress)
 {
-  NarrowString<0x200> url(settings.default_url);
-  url += "/igcfile";
-
   CurlEasy easy{url};
   Curl::Setup(easy);
   const Net::ProgressAdapter progress_adapter{easy, progress};
-  easy.SetFailOnError();
+  // easy.SetFailOnError disabled: HTTP errors are dealt with here at the end
 
-  const auto mime = MakeUploadFlightMime(easy.Get(), settings,
+  const auto mime = MakeUploadFlightMime(easy.Get(), pilot,
                                          glider_type, igc_path);
   easy.SetMimePost(mime.get());
 
   Json::ParserOutputStream parser;
   const auto response =
     co_await Curl::CoStreamRequest(curl, std::move(easy), parser);
-  co_return parser.Finish();
+  auto json =  parser.Finish();
+  std::string error_msg = "";
+
+  if (response.status < 400) {  // 4xx and 5xx are fail codes
+    co_return json;  // return with the WeGlide upload json object
+  } else {
+    // a http error occurs, but not in curl - no SetFailOnError()
+    switch (response.status) {
+    case 400: error_msg = "HTTP failure code: 400 Bad Request:\n";
```

Comment:
Which user will understand the HTTP failure code? If you decide to specifically handle this code by converting it to a user-readable string, there's no point in having this part.
And exception strings should not have newlines. And the two colons look ugly.

Reply:
From my point of view he HTTP failure code is very common also by every user: In case of `SetFailtOnError()` Curl itself gives this code in the error string, every browser shows this code to the user,...
'My' interpretation here is only a guess for the most probably reason for this failure code - and so I also would like to show the curl message too!

Reply:
The CURL error message is just a fallback message, lacking any other means of obtaining error details.
If you don't know for sure what error happened, stop guessing. If you know for sure, explain it in human-readable form, and no HTTP status code is necessary.

Reply:
Maybe, this isn't my opion - but you are the maintainer of all this stuff...

Reply:
No matter whether you want to improve error messages some day; let's skip this part now, and use the standard CURL error handling and its error messages. First get the code in a good shape so it can be merged; error messages can be improved later. Don't make it too hard for yourself.

Reply:
Ok, so I should switch on the `SetFailOnError()` - but very reluctantly: It happens very often to upload a file which is alredy there - and then i got the error code 406. 
Please let me try one more time before I give up...

Reply:
No, forgot my try...  
I will following you proposal to make all changes minimal to finishing this 'basic' PR. Later I will comming back with my update wishes...